File: SelfDividingNumbers.py
from ast import List


class Solution:
  def selfDividingNumbers(self, left: int, right: int) -> List[int]:
    results = []
    for num in range(left, right + 1):
      original = num
      isSelfDividing = True
      while original > 0 and isSelfDividing:
        digit = original % 10
        if digit == 0 or num % digit != 0:
          isSelfDividing = False
        original //= 10
      if isSelfDividing:
        results.append(num)
    return results

File: SelfDividingNumbers.java
import java.util.ArrayList;
import java.util.List;

class Solution {
  public List<Integer> selfDividingNumbers(int left, int right) {
    List<Integer> results = new ArrayList<>();
    for (int num = left; num <= right; num++) {
      int original = num;
      boolean isSelfDividing = true;
      while (original > 0 && isSelfDividing) {
        int digit = original % 10;
        if (digit == 0 || num % digit != 0) {
          isSelfDividing = false;
        }
        original /= 10;
      }
      if (isSelfDividing) {
        results.add(num);
      }
    }
    return results;
  }
}

// Test cases
// left = 1, right = 22

// left = 47, right = 85